mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/AnnotationHandlerMapping.java

```diff
@@ -21,9 +27,45 @@ public AnnotationHandlerMapping(final Object... basePackage) {
 
     public void initialize() {
         log.info("Initialized AnnotationHandlerMapping!");
+        Reflections reflections = new Reflections(basePackage);
+
+        Set<Class<?>> typesAnnotatedWith = reflections.getTypesAnnotatedWith(Controller.class);
+
+        for (Class<?> aClass : typesAnnotatedWith) {
+
+            Object handler;
+            try {
+                handler = aClass.getDeclaredConstructor().newInstance();
+            }catch (Exception e) {
+                log.info("handler 생성 실패!");
+                throw new RuntimeException(e);
+            }
+
+            Method[] declaredMethods = aClass.getDeclaredMethods();
+            for (Method declaredMethod : declaredMethods) {
+                RequestMapping annotation = declaredMethod.getAnnotation(RequestMapping.class);
```

Comment:
코드를 전체적으로 이해 못했을 수 있지만, NPE 문제 위험성이 있어보여요!

```suggestion
                RequestMapping annotation = declaredMethod.getAnnotation(RequestMapping.class);
                if (annotation == null) {
                    continue;
                }
```

방어 로직은 어떤가요?!